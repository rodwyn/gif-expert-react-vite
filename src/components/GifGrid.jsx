import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifItem } from './GifItem';
import PropTypes from 'prop-types';

export const GifGrid = ({category}) => {
  const {images, isLoading} = useFetchGifs(category);

  return (
    <>
        <h1>{category}</h1>
        {
          isLoading && <h2>Cargando...</h2>
        }
        
        <div className='card-grid'>
          {
            images.map((image) => (
              <GifItem key={image.id} {...image}  />
            ))
          }
        </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
};
