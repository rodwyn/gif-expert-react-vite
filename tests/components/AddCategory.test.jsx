import { fireEvent, render, screen } from "@testing-library/react";
import { AddCategory } from "../../src/components/AddCategory";

describe('Testing <AddCategory /> component.', () => {
  test('should render correctly.', () => { 
    const {container} = render(<AddCategory onNewCategory={() => {}} />);    
    expect(container).toMatchSnapshot();
  });

  test('should change input text value.', () => {
    render(<AddCategory onNewCategory={() => {}} />);
    const input = screen.getByRole('textbox');
    fireEvent.input(input, {
        target: {
            value: 'pikachu'
        }
    })

    expect(input.value).toBe('pikachu')
  });

  test('should call onNewCategory if input has value.', () => { 
    const inputValue = 'pikachu';
    const onNewCategory = jest.fn();
    render(<AddCategory onNewCategory={ onNewCategory} />);
    const input = screen.getByRole('textbox');
    const form = screen.getByRole('form');

    fireEvent.input(input, {
      target: { value: inputValue }
    });
    fireEvent.submit(form);
    
    expect(input.value).toBe('');
    expect(onNewCategory).toHaveBeenCalled();
    expect(onNewCategory).toHaveBeenCalledWith(inputValue);
  });

  test('should not call onNewCategory if input has not value.', () => {
    const onNewCategory = jest.fn();
    render(<AddCategory onNewCategory={ onNewCategory} />);
    const form = screen.getByRole('form');
    fireEvent.submit(form);

    expect(onNewCategory).not.toHaveBeenCalled();
  })
});
