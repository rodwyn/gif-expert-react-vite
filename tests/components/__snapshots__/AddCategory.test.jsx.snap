// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <AddCategory /> component. should render correctly. 1`] = `
<div>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
</div>
`;
